### ISSUE TITLE
Single Resistor Design

### ISSUE THREAD

**Issue by Daandreev** [G42makes/ATTiny-i2c-RGB#1]
When the red LED is pulled high there is no current going through the green and blue leds because they have a different voltage drop associated with their diode junction. Effectively all the current passes through the red LED.
![img_1159](https://user-images.githubusercontent.com/12501592/43017619-9e4e533e-8c0b-11e8-8f76-8c84d3de0837.JPG)

**Comment by G42makes** [comment]
Makes sense, thanks.

I really should get this done up as small PCB project with some surface mount resistors, I could easily balance the 3 colours then, but mostly by looking up the values for them.

I need to spend some time one day learning the basic maths for electronics that you did here.

I probably won't update this project any time soon, unfortunately, so I'll just have to leave this visible if anyone wants to know what's up.

**Comment by Daandreev** [comment]
I'm about to make a very similar project and plan on using your library/code; Will connect an esp8266 that is sitting on a meshnet to a attiny85 which will handle LED dimming.

**Comment by G42makes** [comment]
Surprisingly similar to what I was doing, I literally wrote this to be controlled from an esp8266.

I have a design that allowed me to do HID over ethernet(in a very ghetto design), but it mostly worked, still had some timing issues. That would be embedded in the NES, inline with the controller ports to allow me to send gamepad commands from another esp8266 in the controller, wirelessly. I had this support colour changes from the esp8266 so that I could use a web interface to change it as I pleased.

Mostly was working, but I stalled out on the project due to time constraints.

Feel free to use the code, and have fun.
